sdks/python/apache_beam/runners/worker/sdk_worker.py

```diff
@@ -305,10 +316,14 @@ def get_bundle_processor(self, instruction_id, bundle_descriptor_id):
       with state_handler.process_instruction_id(instruction_id):
         yield processor
     finally:
-      del self.active_bundle_processors[instruction_id]
+      requires_finalization = \
+        self.active_bundle_processors[instruction_id].requires_finalization
```

Comment:
OK, I'm seeing why this is an instance variable now. It's odd to "return" whether finalization is required both as a self mutation and in the return value--how about we make this a method (it should be really cheap to compute; probably don't need to cache the result) and use that uniformly everywhere?